hi-ml/src/health_ml/model_trainer.py

```diff
@@ -0,0 +1,199 @@
+                         f"To change, modify the pl_progress_bar_refresh_rate field of the container.")
+        callbacks.append(AzureMLProgressBar(refresh_rate=progress_bar_refresh_rate,  # type: ignore
+                                            write_to_logging_info=True,
+                                            print_timestamp=False))
+    trainer = Trainer(default_root_dir=str(container.outputs_folder),
+                      deterministic=deterministic,
+                      benchmark=benchmark,
+                      accelerator=accelerator,
+                      plugins=plugins,
+                      max_epochs=container.max_epochs,
+                      # Both these arguments can be integers or floats. If integers, it is the number of batches.
+                      # If float, it's the fraction of batches. We default to 1.0 (processing all batches).
+                      limit_train_batches=container.pl_limit_train_batches or 1.0,
+                      limit_val_batches=container.pl_limit_val_batches or 1.0,
+                      num_sanity_val_steps=container.pl_num_sanity_val_steps,
+                      callbacks=callbacks,
+                      logger=loggers,
+                      progress_bar_refresh_rate=progress_bar_refresh_rate,
+                      num_nodes=num_nodes,
+                      gpus=num_gpus,
+                      precision=precision,
+                      sync_batchnorm=True,
+                      terminate_on_nan=container.detect_anomaly,
+                      profiler=container.pl_profiler,
+                      resume_from_checkpoint=str(resume_from_checkpoint) if resume_from_checkpoint else None,
+                      **kwargs)
+    return trainer, storing_logger
+
+
+def model_train(container: LightningContainer,
+                num_nodes: int = 1) -> Tuple[Trainer, StoringLogger]:
+    """
+    The main training loop. It creates the Pytorch model based on the configuration options passed in,
+    creates a Pytorch Lightning trainer, and trains the model.
+    If a checkpoint was specified, then it loads the checkpoint before resuming training.
+
+    :param num_nodes: The number of nodes to use in distributed training.
+    :param container: A container object that holds the training data in PyTorch Lightning format
+    and the model to train.
+    :return: A tuple of [Trainer, StoringLogger]. Trainer is the Lightning Trainer object that was used for fitting
+    the model. The StoringLogger object is returned when training a built-in model, this is None when
+    fitting other models.
+    """
+    lightning_model = container.model
+
+    # resource_monitor: Optional[ResourceMonitor] = None
+    # Execute some bookkeeping tasks only once if running distributed:
+    if is_global_rank_zero():
+        logging.info(f"Model checkpoints are saved at {container.checkpoint_folder}")
+        write_args_file(container,
```

Comment:
this "write_args_file" is somewhat of a misnomer. Can we call it "model summary" or something alike? "experiment summary"?